src/vsr/replica.zig

```diff
@@ -4884,8 +4884,18 @@ pub fn ReplicaType(
                 op_head_no_gaps,
             );
             const op_min = if (range) |r| r.op_max + 1 else range_min;
+
             assert(op_min <= op);
-            assert(op_min <= self.op_repair_min());
+            if (self.op_checkpoint() == 0 and range != null) {
+                // We may get here with a missing root op if a we're a backup that is advancing its
+                // checkpoint mid repair.
+                assert(self.status == .normal and self.backup());
+                assert(self.commit_min == self.op_checkpoint_next_trigger());
+                assert(range.?.op_max == 0);
+                assert(range.?.op_min == 0);
```

Comment:
> Though, I'm missing how we know that range.min == range.max == 0 here.

@sentientwaffle Mhmmmm, this flow _is_ getting a bit confusing. 😕 

`update_start_view_headers` is called by either:
1. primary about to perform superblock.checkpoint
2. primary refreshing its SV headers to send out start_view
3. backup about to perform superblock.checkpoint

For cases 1 & 2, the primary is _guaranteed_ to have a clean op=0 while invoking `update_start_view_headers` (repair ensures that). Even if op=0 is corrupted during normal processing, it can't be marked as faulty as it is never read from disk at runtime (we start committing from op=1).
https://github.com/tigerbeetle/tigerbeetle/blob/ed0a1c73a24b2caa9fbcecf54b52dfbb27ded83d/src/vsr/replica.zig#L6473

For case 3, however, a backup could be repairing and advancing its checkpoint while it has a corrupt op=0 (if it is read as corrupted on startup).

Reply:
@sentientwaffle: Although, your comment pushed me to think harder about this entire situation and now I feel that this solution is **_VERY_** wrong and could certainly have gaps other than in slot=0 here... 

We can only guarantee here we have no breaks between op_checkpoint → op_head_no_gaps. Here, since `range_min=op_head_no_gaps + 1) -| constants.journal_slot_count;`, it could actually _precede_ op_checkpoint! And we actually have no guarantees about the journal being clean from range_min → op_checkpoint.
https://github.com/tigerbeetle/tigerbeetle/blob/ed0a1c73a24b2caa9fbcecf54b52dfbb27ded83d/src/vsr/replica.zig#L4872-L4878

I am also very surprised at the fact that VOPR1000 hasn't been able to find a failing seed related to this. In fact, the PR is doing quite well CFO. 😅